game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/StrengthAndRollCalculator.java

```diff
@@ -13,8 +13,18 @@
   // Keep track of the units that are providing support and whom they are providing it to
   @Getter Map<Unit, IntegerMap<Unit>> supportGiven = new HashMap<>();
 
-  int addSupport(final Unit unit, final AvailableSupports supportTracker) {
-    final IntegerMap<Unit> supportGivenToUnit = supportTracker.giveSupportToUnit(unit);
+  /**
+   * Gives support to the unit and removes it from the availableSupport
+   *
+   * <p>Each time this is called, the amount of available support will decrease equal to the amount
+   * returned.
+   *
+   * @param unit Unit to have support given to it
+   * @param availableSupport Keeps state of what support is still available
+   * @return the amount of support given
```

Comment:
might be worthwhile clarifying if the support is a strength bonus or a roll bonus. Is there a way to know which one will be returned?

Reply:
There isn't a way to determine if it is a strength or roll bonus.  That is controlled by the list of support rules in AvailableSupports.

I just removed this file, though.  The logic that it had looked like it fit better in AvailableSupports.  So now the `supportGiven` map is managed in `AvailableSupports` and the `CombatValue` classes will merge the two `supportGiven` maps from the friendly and enemy supports.